methods/inputs/locate_gedi_data.py

```diff
@@ -103,48 +124,106 @@ def gedi_fetch(boundary_file: str, gedi_data_dir: str) -> None:
+                date_range=(dt.datetime(start_year, 1, 1), dt.datetime(end_year, 1, 1)),
                 spatial=shape
             )
             granule_metadatas.append(result)
 
+            # move to the next chunk
             chunk = chunked_layer.GetNextFeature()
 
+    # concatenate all metadata into a single dataframe and remove duplicates
     granule_metadata = pd.concat(granule_metadatas).drop_duplicates(subset="granule_name")
 
+    # write the granule names to a text file in the specified directory
+    granule_names = granule_metadata["granule_name"].tolist()
+    output_txt_path = os.path.join(output_txt_dir, "gedi_names.txt")
+    with open(output_txt_path, "w") as txt_file:
+        for name in granule_names:
+            txt_file.write(f"{name}\n")
+
+    # save each granule's metadata as a JSON file in the specified GEDI data directory
     for _, row in granule_metadata.iterrows():
         basename, _ = os.path.splitext(row.granule_name)
         with open(os.path.join(gedi_data_dir, f"{basename}.json"), "w", encoding='utf-8') as f:
-            f.write(json.dumps(
+            json.dump(
                 {
                     "name": row.granule_name,
                     "url": row.granule_url
-                }
-            ))
+                },
+                f
+            )
 
+
+# main function to handle input arguments and call the data-fetching function
 def main() -> None:
-    try:
-        boundary_file = sys.argv[1]
-        gedi_data_dir = sys.argv[2]
-    except IndexError:
-        print(f"Usage: {sys.argv[0]} BUFFER_BOUNDRY_FILE GEDI_DATA_DIRECTORY")
-        sys.exit(1)
-    except DownloadError as exc:
-        print(f"Failed to download: {exc.msg}")
-        sys.exit(1)
-
-    gedi_fetch(boundary_file, gedi_data_dir)
+    # set up argument parsing
+    parser = argparse.ArgumentParser(
+        description="Fetch GEDI data for the given boundary and save metadata as JSON and text files."
+    )
+
+    # add argument for the GEDI data directory where JSONs will be stored
+    parser.add_argument(
+        "--gedi-dir",
+        type=str,
+        required=True,
+        help="Directory to store the GEDI JSON data files"
+    )
+
+    # add argument for the boundary file (e.g., GeoJSON, shapefile)
+    parser.add_argument(
+        "--buffer",
+        type=str,
+        required=True,
+        help="Path to the buffer boundary file"
+    )
+
+    # add argument for the output folder to store the text file with granule names
+    parser.add_argument(
+        "--output-folder",
+        type=str,
+        required=True,
+        help="Output folder where the text file with granule names will be saved"
+    )
+
+    # add arguments for the start year and end year
+    parser.add_argument(
+        "--start-year",
+        type=int,
+        required=True,
+        help="Start year for querying GEDI data (inclusive)"
+    )
+
+    parser.add_argument(
+        "--end-year",
```

Comment:
Why is this now an argument, rather than a constant, when in generate_carbon_density.py the JRC tile year is still fixed? 

I guess the concern here is that if this is configurable, then mistakes will happen :)  But if we do want it to be configurable (as it's now part of the methodology) then we need to update the generate_carbon_density.py script to load the correct tiles, and indeed load the correct years for the correct GEDI samples.

Reply:
Ah I see yes, I'll revert it back to having a fixed time-frame (that is now double validated in both locate and filter)